src/mavsdk/plugins/action/action_impl.cpp

```diff
@@ -450,6 +450,11 @@ void ActionImpl::goto_location_async(
 
             command.command = MAV_CMD_DO_REPOSITION;
             command.target_component_id = _system_impl->get_autopilot_id();
+            // When ArduPilot we need to set frame to Global i.e. positive altitude over mean sea
+            // level (MSL)
+            if (_system_impl->autopilot() != SystemImpl::Autopilot::Px4) {
```

Comment:
Does it make a difference for PX4? I'm thinking that maybe we could just drop that `if` and set `MAV_FRAME_GLOBAL` for both PX4 and Ardupilot?

Probably a question for @julianoes :innocent:

Reply:
We are sending a COMMAND_INT, so I would argue it should be `MAV_FRAME_GLOBAL_INT`. Objections?

And from what I can see PX4 ignores the frame, so we might as well set it as it's the right thing to do.

Reply:
@paulholmstedt: Can you confirm that Ardupilot is fine with `MAV_FRAME_GLOBAL_INT`? :innocent: